src/BuildScriptGeneratorCli/readme.md

```diff
@@ -0,0 +1,18 @@
+# Oryx Cli Structure
+## Background
+Due to the transition to net7, we found that the past cli package isn't compatible with the new framework. Hence, we've decided to move to `System.CommandLine` package to develop our cli. For developers, the structure of the cli codes are different then the past.
+
+## Structure
+- Program.cs (Define root command `oryx` and add subcommands)
+  - Subcommand.cs (Define subcommands, register its options and arguments, and define the execute action)
+
+### 1. Command Files
+Now, the command folder will not handle option and argument inputs directly. This will be done in the binder. The command file will only define the subcommands, register its options and arguments, and also define the execute action. The command (with all its options and executes) will then be exported to be added to the root command.
+
+**Please be aware, developer still need to define the properties used in the execute method in command file!** The property file is a intermediate step in order to make the binder's result accessable to the command file.
```

Comment:
nit: "developer still" --> "the developer will still", "in the execute method" --> in the `Execute` method"